tests/Dialect/RNS/IR/syntax.mlir

```diff
@@ -22,3 +22,32 @@ func.func @test_syntax(%arg0: !ty) -> !ty {
 !poly_ty_bad = !polynomial.polynomial<ring=#ring_bad>
 // expected-error@+1 {{RNS type has incompatible basis types}}
 !ty_bad = !rns.rns<!poly_ty_1, !poly_ty_2, !poly_ty_bad>
+
+// -----
+// mod arith
+
+!Zp1 = !mod_arith.int<3721063133 : i64>
+!Zp2 = !mod_arith.int<2737228591 : i64>
+!Zp3 = !mod_arith.int<3180146689 : i64>
+
+!ty_modarith = !rns.rns<!Zp1, !Zp2, !Zp3>
+
+func.func @test_syntax_modarith(%arg0: !ty_modarith) -> !ty_modarith {
+  return %arg0 : !ty_modarith
+}
+
+// expected-error@+1 {{RNS type has incompatible basis types}}
+!ty_modarith_bad = !rns.rns<!Zp1, !Zp2, !Zp1>
+
+// -----
+
+!Zp1 = !mod_arith.int<3721063133 : i64>
+!Zp2 = !mod_arith.int<65537 : i32>
+
+// expected-error@+1 {{RNS type has incompatible basis types}}
+!ty_modarith_bad = !rns.rns<!Zp1, !Zp2>
+
+// -----
+
+// expected-error@+1 {{does not have RNSBasisTypeInterface}}
+!ty_int_bad = !rns.rns<i32, i64>
```

Comment:
Please add a test where it succeeds as well.

Reply:
I think there is no case for i32 to pass the check?

Reply:
Sorry, I meant for mod_arith types. All tests added in this PR, even for mod arith, check error cases.

Reply:
Added a succeeding case for storage type check.